src/pytopmod/core/mesh.py

```diff
@@ -0,0 +1,39 @@
+import dataclasses
+
+from pytopmod.core.face import FaceKey
+from pytopmod.core.geometry import Point3D
+from pytopmod.core.keystore import KeyStore
+from pytopmod.core.vertex import VertexKey
+
+
+@dataclasses.dataclass(slots=True)
+class Mesh:
+    """Base Mesh class, to be subclassed (e.g DLFLMesh, DCELMesh, ect.)
+
+    Stores and provides create/delete methods for vertice and face keys.
+    Also exposes a vertex coordinates map.
+    """
+
+    vertices: KeyStore[VertexKey] = dataclasses.field(init=False)
+    faces: KeyStore[FaceKey] = dataclasses.field(init=False)
+    vertex_coordinates: dict[VertexKey, Point3D] = dataclasses.field(init=False)
```

Comment:
I may not be reading the Python implementation correctly, but it looks like we are treating the vertex and its coordinates as two separate entities. What is the reason for that?

Reply:
yes, from what I understand from the code:
- vertices would be a list like ["v1", "v2", "v3"]
- faces would be like ["f1", "f2", "f3", "f4"]
- and then vertex_coordinates would be something like {"v1": (1.0, 1.0, 1.0), "v2": (1.0, -1.0, -1.0), "v3": (-1.0, 1.0, -1.0)}

so then you can use those vertices and faces "keys" in operations like this:
    operators.insert_edge(mesh, "v1", "f1", "v2", "f2")
    operators.insert_edge(mesh, "v2", "f5", "v3", "f3")

Reply:
Sorry for the late reply, but yes that's exactly that, we are using simple string keys for all entities and storing their properties in separate structures so we can easily manipulate them (without having to do a lookup).

Reply:
Marking as resolved